code/obj/machinery/singularity.dm

```diff
@@ -639,43 +639,19 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 			return
 
 	if(isweldingtool(W))
-
-		var/turf/T = user.loc
-
+		var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(user, src, 2 SECONDS, /obj/machinery/field_generator/proc/weld_action,\
```

Comment:
why not use the generic one

Reply:
I'm using `actions.start` twice per actionbar definition. I figured the code is cleaner that way

Reply:
@Adharainspace variant where it doesn't auto start, and returns the ref?

Reply:
okay, I actually managed to make it work with single generic actionbar per emitter/field gen